docs/devel/api_changes.md

```diff
@@ -0,0 +1,237 @@
+# So you want to change the API?
+
+The Kubernetes API has two major components - the internal API structures and
+the versioned API structures.  The versioned APIs are intended to be stable,
+while the internal API is implemented to best reflect the needs of the
+Kubernetes code itself.  Clients should consume the versioned APIs exclusively.
+
+What this means for API changes is that you have to be somewhat thoughtful in
```

Comment:
It would be useful to say more about the relationship between the internal rep and the versioned APIs -- we translate all versions to a common internal representation. We should explain Decode/Encode: parsing, defaulting, conversion, validation, etc.